packages/app-mobile/components/NoteList.tsx

```diff
@@ -91,7 +92,13 @@ class NoteListComponent extends Component<NoteListProps> {
 				keyExtractor={item => item.id}
 			/>;
 		} else {
-			if (!Folder.atLeastOneRealFolderExists(this.props.folders)) {
+			const notesSourceParentId = this.props.notesSource ? JSON.parse(this.props.notesSource).parentId : null;
+			const notesSourceFolder = this.props.folders.find(folder => folder.id === notesSourceParentId);
+
+			if (itemIsInTrash(notesSourceFolder)) {
```

Comment:
I don't get what this JSON parsing is doing here. It's not a pattern that appears anywhere else in the code as far as I know? Wouldn't props.selectedFolderId give you what you want?

Reply:
The `notesSource` prop is a json string which consists of the selected folder's id as `parentId` attribute,

![image](https://github.com/laurent22/joplin/assets/97161173/83de52ba-bd6c-491f-8a28-f261072f6bc7)
![image](https://github.com/laurent22/joplin/assets/97161173/06f94b24-0e96-4aec-b42a-6ff841fa3ce6)
so I parsed it got the folder id.

`NoteList` didn't have the `props.selectedFolderId` , so I used  `notesSource` instead of adding it from the redux state.

Do you suggest I add and use `props.selectedFolderId` instead?

Reply:
I have tried using `props.selectedFolderId`, it is failing in some cases. When there are no notebooks and the app opens to the "all notes" screen, the initial `selectedFolderId` is null. Then if we go to trash folder, it is set to the trash folder id. Now, if we go back to "all notes" screen, the value of the `selectedFolderId` still remains the same as the trash folder id and the "There are no notes here." message is shown. This doesn't occur when using the `parentId` from the `notesSource` as it updates to the `ALL_NOTES_FILTER_ID` when we go back to "all notes" screen.

Reply:
Based on your comment I would prefer not to merge this, as this confusion with trash folder ID and all notes need to be fixed first.

Reply:
I have created an issue about it here: https://github.com/laurent22/joplin/issues/10259